include/boost/url/detail/impl/any_segments_iter.ipp

```diff
@@ -120,20 +120,24 @@ void
 path_iter::
 copy(
     char*& dest,
-    char const* end) noexcept
+    char const* end,
+    bool encode_colons) noexcept
 {
     BOOST_ASSERT(pos_ !=
         string_view::npos);
     encode_opts opt;
     opt.space_to_plus = false;
+    auto cs = pchars;
+    if (encode_colons)
+        cs = cs - ':';
```

Comment:
There needs to be two character set constants, constexpr, otherwise we are running code needlessly